src/pages/billing/Project.js

```diff
@@ -22,44 +22,82 @@ export default ajaxCaller(class ProjectDetail extends Component {
+      }
+    })
+
+  refresh = withErrorReporting('Error loading billing project users list',
+    async () => {
+      const { ajax: { Billing }, project } = this.props
+      this.setState({ addingUser: false, deletingUser: false, updating: false, editingUser: false })
+      const rawProjectUsers = await Billing.project(project.projectName).listUsers()
+      const projectUsers = _.flow(
+        _.groupBy('email'),
+        _.map(gs => ({ ..._.omit('role', gs[0]), roles: _.map('role', gs) })),
+        _.sortBy('email')
+      )(rawProjectUsers)
+      this.setState({ projectUsers })
+    })
+
+  componentDidMount = Utils.withBusyState(
+    loading => this.setState({ loading }),
+    () => Promise.all([
+      this.refresh(),
+      this.loadBillingInfo()
+    ])
+  )
 
   render() {
-    const { project: { projectName, creationStatus }, ajax: { Billing } } = this.props
-    const { projectUsers, loading, updating, filter, addingUser, deletingUser, editingUser } = this.state
+    const { project: { projectName, creationStatus }, ajax: { Billing }, billingAccounts, authorizeAndLoadAccounts, loadingAccountAuth } = this.props
+    const { projectUsers, loading, updating, filter, addingUser, deletingUser, editingUser, billingAccountName } = this.state
     const adminCanEdit = _.filter(({ roles }) => _.includes('Owner', roles), projectUsers).length > 1
+    const billingNode = Utils.cond(
+      [!!billingAccounts, () => h(Select, {
+        value: billingAccountName,
+        isClearable: false,
+        styles: { container: old => ({ ...old, width: 320 }) },
+        options: _.map(({ displayName, accountName }) => ({ label: displayName, value: accountName }), billingAccounts),
+        onChange: ({ value: newAccountName }) => this.updateBillingAccount(newAccountName)
+      })],
+      [!billingAccounts && !loadingAccountAuth, () => buttonPrimary({
+        onClick: async () => {
+          await authorizeAndLoadAccounts()
+          Utils.withBusyState(loading => this.setState({ loading }), this.loadBillingInfo)()
+        }
+      }, 'Authorize')]
+    )
 
     return h(Fragment, [
       div({ style: { padding: '1.5rem 3rem', flexGrow: 1 } }, [
-        div({ style: { color: colors.dark(), fontSize: 16, fontWeight: 600 } }, [
+        div({ style: { color: colors.dark(), fontSize: 16, fontWeight: 600, display: 'flex', alignItems: 'center' } }, [
           projectName,
           span({ style: { fontWeight: 500, fontSize: 14, margin: '0 1.5rem 0 3rem' } }, creationStatus),
           Utils.cond(
             [creationStatus === 'Ready', () => icon('check', { style: { color: colors.success() } })],
             [creationStatus === 'Creating', () => spinner({ size: 16 })],
             () => icon('error-standard', { style: { color: colors.danger() } })
-          )
+          ),
+          billingNode && span({ style: { flexShrink: 0, fontWeight: 500, fontSize: 14, margin: '0 0.75rem 0 auto' } }, 'Billing Account For This Project:'),
```

Comment:
Why is `billingNode` used as a conditional here?

Reply:
Originally, I only wanted the label/span to display once the type of node was determined (which was async). I now use hasBillingScope for this which is synchronous - thanks!